### PR TITLE
[#9382] Refactor instructor course edit page

### PR DESCRIPTION
**Pull request by joanneong**
Part of #9382 

As per the discussion from #9429 , the instructor course edit page has been refactored to reduce duplicated code and direct DOM manipulations.

Here are some comments regarding this PR:

~1. I did not fully reuse/ add on to existing interfaces~
   ~- There is an incongruence between the course defined in the web interface (which uses the attribute names `courseId` and `courseName`) and `CourseAttributes` (which uses the attribute names `id` and `name`). This means that when I tried to get a Course attributes from the backend and access it as a `Course`, I could not use the same terms to access the course id and name. Renaming either the web interface or the course attributes can potentially affect many files, so I just defined another instance in instructors edit page component.~

   ~- I have not reused instructor privileges as my definitions are somewhat different and refactoring the current instructor privileges file (`instructor-privilege.ts`) will affect many other files.~

2. There is possibly a logic bug in editing instructors
   - There seems to be a bug in the logic regarding displayed instructors for editing instructors. When you attempt to edit an instructor such that there will be just one instructor displayed to students in a course, the error message "At least one instructor must be displayed to students" is shown. However, this problem does not happen when you try to add an instructor that is not displayed to students.

3. I have yet to delete old files
   - Just to clarify, should I only delete files without any reference outside of instructor course edit page? For example, `InstructorCourseEditPage.java` is involved in a UI test for the admin home page, so deleting it will mean that I have to modify files not directly related to the instructor course edit page.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
> This means that when I tried to get a Course attributes from the backend and access it as a `Course`, I could not use the same terms to access the course id and name

Not sure if I get you here.

> I have not reused instructor privileges as my definitions are somewhat different

How are they different?

> I have yet to delete old files

I don't think there are any old files left for you to delete.

**Comment by joanneong** [comment]
While replying to your comment, I relooked the available APIs and managed to reconcile the definitions. Hence problem 1. has been resolved.

**Comment by joanneong** [comment]
Also, I am not too sure about how to modify `api-output.ts` so that it reflects the correct API changes. I tried to change and commit by removing the file from `.gitignore`, but it seems like there are still issues with that...

File: src/main/java/teammates/ui/webapi/output/InstructorData.java [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
+    private final String name;
     private final String courseId;
     private final String email;
-
+    private String role;
     private Boolean isDisplayedToStudents;
-    private final String displayedToStudentsAs;
-    private final String name;
+    private final String displayedName;
+    private InstructorPrivileges privileges;
```
Comment (wkurniawan07) [inline src/main/java/teammates/ui/webapi/output/InstructorData.java]: @joanneong Check all the places where this DTO is used as the output. In those places, if any of the information is not supposed to be visible, you should handle accordingly.

Comment (joanneong) [reply src/main/java/teammates/ui/webapi/output/InstructorData.java]: Can you clarify what you mean here? I don't think I fully understand what you are saying. Do you mean the `InstructorData` DTO?

Comment (wkurniawan07) [reply src/main/java/teammates/ui/webapi/output/InstructorData.java]: Yes. Remember, API responses can be traced even though the web page can decide to filter some information.

File: src/web/app/pages-instructor/instructor-pages.module.ts [hunk 2]
```diff
@@ -204,6 +204,7 @@ const routes: Routes = [
     InstructorCourseStudentDetailsPageModule,
     InstructorCourseStudentEditPageModule,
     InstructorHomePageModule,
+    InstructorCourseEditPageModule,
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-pages.module.ts]: Isn't this already imported?

**Comment by wkurniawan07** [comment]
@joanneong I'm worried that your work here might clash with the API endpoints cleanup that's currently ongoing. Do you absolutely need to modify any API endpoint? I am under the impression that only the front-end needs change.

**Pull request closed without merging**